client/scripts/components/Room/Room.js

```diff
@@ -71,10 +72,23 @@ class Room extends Component {
 
     this.dismissWelcomeModal = () => {
       if (this.props.myNametag) {
-        this.setState({dismissWelcomeModal: true})
+        this.setState({dismissedWelcomeModal: true})
       }
     }
 
+    this.handleRoomJoin = () => {
+      // FIXME: We set hasPosted=false because there's a gap in between nametag
```

Comment:
One nice way to handle this might be to check `this.props.data.loading`, which should be `false` once messages are loaded. This is how I've managed null states during loading in the rest of the application.